File: event.py
class Fight:
    def __init__(self, id, fighter1, fighter2, fighter1_won):
        self.id = id
        self.fighter1 = fighter1
        self.fighter2 = fighter2
        self.fighter1_won = fighter1_won

    def __str__(self):
        return(f"fighter1: {self.fighter1} fighter2: {self.fighter2}")
    
    def to_json(self):
        return {"fighter1": self.fighter1, "fighter2": self.fighter2, "fighter1_won": self.fighter1_won}

class Event:
    def __init__(self, id, name, date, location):
        self.id = id
        self.name = name
        self.date = date
        self.fights = []
        self.location = location

    def to_json(self):
        fights_dict = [fight.to_json() for fight in self.fights]
        return {"id": self.id, "name": self.name, "date": self.date, 
                "location": self.location, "fights": fights_dict}

    def __str__(self):
        string = f"date: {self.date} event: {self.name}\n"
        for fight in self.fights:
            string = string + str(fight) + "\n"
        return(string)

File: event.js
const apiUrl = 'http://127.0.0.1:5000';
const eventEndpoint = `${apiUrl}/events`
events = [];

fetch(eventEndpoint)
    .then((response) => response.json())
    .then((events) => {
        displayUFCEvents(events);
    })

function displayUFCEvents(events) {
  try {
        // Get the table body element
        const tbody = document.querySelector('#ufc-events-table tbody');

        // Clear existing rows
        tbody.innerHTML = '';

        // Populate the table with event data
        events.forEach(event => {
            const row = document.createElement('tr');
            const date = new Date(event.date * 1000);
            const month = date.toLocaleString('default', {month: 'long'});
            const day = date.getDate();
            console.log(day);
            const year = date.getFullYear();

            const formattedDate = `${month}-${day}-${year}`;

            row.innerHTML = `
                <td><a href="detail.html?id=${event.id}">${event.name}</td>
                <td>${date.toDateString()}</td>
                <td>${event.location}</td>
                <td><a href="https://watchmmafull.com/search/${formattedDate}/"target="_blank" class="watch-icon">🎥</a></td>
            `;

            tbody.appendChild(row);
        });
    } catch (error) {
        console.error('Error fetching UFC events:', error);
    }   
}

window.onload = displayUFCEvents(events);
